ocs_ci/framework/main.py

```diff
@@ -41,10 +42,26 @@ def init_ocsci_conf(arguments=None):
     if custom_config:
         with open(os.path.expanduser(custom_config)) as file_stream:
             custom_config_data = yaml.safe_load(file_stream)
+            custom_config_data_str = str(custom_config_data)
+            if "'platform': 'vsphere'" in custom_config_data_str:
+                if not os.path.isfile(constants.VSPHERE_CONFIG_PATH):
+                    msg = "vsphere_upi_vars.yaml MUST be provided for vSphere Platform"
+                    pytest.fail(msg)
```

Comment:
I think that in this stage we do not run pytest yet so not sure if pytest.fail will raise the exception or what it exactly does but not sure it's ok when we do not run pytest yet as it's executed on line 81

Reply:
Tested pytest.fail and its fail as expected when we don't provide vsphere_upi_vars.yaml